_posts/2018-05-31-elixir-in-action-review.md

```diff
@@ -0,0 +1,87 @@
+* add a custom server process to maintain state
+* migrate state to a GenServer
+* add a cache so that we start supporting multiple to-do lists
+* add basic data persistence
+* add a supervision tree and introduce a basic pool of workers
+Additionally, we are introduced to ETS tables, OTP Applications and configuration, dependencies and Plug. In this section, I got the feeling that I am pair programming as a junior with the author, sitting beside him while he designs the application. When he finds a hole in his design, he questions himself "What part of the OTP can cover this hole, and how?". Then, he turns to me and gives me a detailed explanation on what we are doing, proceeds with the implementation on the todo-list app, and finally turns to me again and discusses on alternatives and ups and downs of the chosen path. I like this aproach, and I want to see it in more language learning books. 
+
+In the last two chapters, we can see how to make our app run on a cluster, release it using Distillery, and maintain it (Debug, Log etc). The author follows the same approach here as well, albeit not in that much detail, as the topics discussed in this section are very broad (clustering, data replication, network partitions, releases) and would need way too much detail to cover in a language introduction book.
+
+## Conclusion
+
+Lately books are being pushed to the side when it comes to learning new technology and not without good reason, reading and understanding a book requires dedication and a lot of time.
+These days most people don't seem to have the time for books, so instead videos and brief tutorials are used to jump start learning.
+
+Another factor working against books is that many authors omit important steps of their thinking process, which ultimately leaves us, the readers, unable to produce a similar solution ourselves.
+We're left to wonder: "How the heck did we come up with this solution?  What was the mindset that lead us to this point?".
+Fortunately, Saša is excellent at helping the reader understand his reasoning, he explains each step of the process while exploring alternatives when he thinks it's due.
+
+So is Saša Jurić's newest book _Elixir in Action, Second Edition._ worth the investment of your time?
+
+If you're an experienced developer looking to advance your understanding of Elixir and the underlying BEAM: yes.
+
+This IS NOT a beginner's book.
+This IS NOT a simple book.
+This IS an excellent resource highlighting the potential and shortcomings of Elixir and the BEAM.
+
+If you give it your attention you'll finish the book with a solid understanding of the platform and how best to leverage it at all stages of the development lifecycle.
+
+Look for more reviews to come soon!
```

Comment:
How do you feel about us redoing the bottom portion here?  Additionally we can edit the post to include the specific twitter URL after we send it out.

```markdown
## Giveaway

To kick off our new series we are giving away free copies of Saša Jurić's _Elixir in Action, Second Edition._!

Want your chance to win a free copy?
Follow us on [Twitter](https://twitter.com/elixirschool) and retweet this blog post announcement!
We'll be picking winners at the end of the month, don't miss your chance!

Look for more reviews to come soon!
```